### PR TITLE
Improve audio

### PR DESCRIPTION
**Pull request by rshym**
Extended supported sample rates.
Improve error reporting.

### CODE REVIEW

File: stream_in.c [hunk 1]
```diff
@@ -397,12 +397,13 @@ int in_create(struct audio_hw_device *dev,
...
         ALOGE("%s() failed (%d). Can't open stream on device.", __FUNCTION__, errno);
         if (xin->p_handle != NULL) {
+            ALOGE("pcm error:'%s'", pcm_get_error(xin->p_handle));
             pcm_close(xin->p_handle);
             xin->p_handle = NULL;
         }
         *stream_in = NULL;
         free(xin);
-        return -ENOMEM;
+        return -errno;
```
Comment (arminn) [inline stream_in.c]: Consider the possibility of changing errno by free()

File: stream_out.c [hunk 2]
```diff
@@ -467,12 +467,13 @@ int out_create(struct audio_hw_device *dev,
...
         ALOGE("%s() failed (%d). Can't open stream on device.", __FUNCTION__, errno);
         if (xout->p_handle != NULL) {
+            ALOGE("pcm error:'%s'", pcm_get_error(xout->p_handle));
             pcm_close(xout->p_handle);
             xout->p_handle = NULL;
         }
         *stream_out = NULL;
         free(xout);
-        return -ENOMEM;
+        return -errno;
```
Comment (arminn) [inline stream_out.c]: Consider the possibility of changing errno by free()

File: stream_in.c (hunk 1 above)
Comment (rshym) [reply stream_in.c]: Thanks. Fixed.

File: stream_out.c (hunk 2 above)
Comment (rshym) [reply stream_out.c]: Thanks. Fixed.

**Comment by arminn** [comment]
Reviewed-by: Andrii Chepurnyi andrii_chepurnyi@epam.com

**Comment by rshym** [comment]
Merged as part of #8.

**Pull request closed without merging**